packages/astro/src/core/app/index.ts

```diff
@@ -160,8 +162,19 @@ export class App {
 		);
 		let response;
 		try {
-			if (mod.onRequest) {
-				this.#pipeline.setMiddlewareFunction(mod.onRequest as MiddlewareEndpointHandler);
+			let i18nMiddleware = createI18nMiddleware(this.#manifest.i18n);
+			if (i18nMiddleware) {
+				if (mod.onRequest) {
+					this.#pipeline.setMiddlewareFunction(
+						sequence(i18nMiddleware, mod.onRequest as MiddlewareEndpointHandler)
+					);
+				} else {
+					this.#pipeline.setMiddlewareFunction(i18nMiddleware);
+				}
+			} else {
+				if (mod.onRequest) {
+					this.#pipeline.setMiddlewareFunction(mod.onRequest as MiddlewareEndpointHandler);
+				}
```

Comment:
Shouldn't there be very similar code in ssg generate and dev? Is the middleware added differently there?

Reply:
Oh, I see what you mean! The only issue I see is that the manifest isn't inside the `Pipeline`, which means that we can't move the `if/else`  blocks to check `manifest.i18n` inside a single place

Reply:
So how *is* the middleware added in those places currently?

Reply:
I think it's ok for now. When we (or I) do the middleware integrations proposal we can revisit.